src/main/kotlin/no/nav/bidrag/bidragskalkulator/mapper/BeregningsgrunnlagBuilder.kt

```diff
@@ -0,0 +1,120 @@
+package no.nav.bidrag.bidragskalkulator.mapper
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import no.nav.bidrag.bidragskalkulator.dto.BarnDto
+import no.nav.bidrag.bidragskalkulator.dto.BidragsType
+import no.nav.bidrag.bidragskalkulator.mapper.BeregningsgrunnlagMapper.Referanser
+import no.nav.bidrag.domene.enums.grunnlag.Grunnlagstype
+import no.nav.bidrag.domene.enums.inntekt.Inntektsrapportering
+import no.nav.bidrag.domene.enums.person.Bostatuskode
+import no.nav.bidrag.domene.tid.ÅrMånedsperiode
+import no.nav.bidrag.transport.behandling.felles.grunnlag.*
+import org.springframework.stereotype.Component
+import java.math.BigDecimal
+import java.time.LocalDate
+import java.time.YearMonth
+
+@Component
+class BeregningsgrunnlagBuilder(
+    private val objectMapper: ObjectMapper = ObjectMapper().registerKotlinModule().registerModule(JavaTimeModule())
+) {
+
+    fun byggPersongrunnlag(referanse: String, type: Grunnlagstype, fødselsdato: LocalDate? = null) = GrunnlagDto(
+        referanse = referanse,
+        type = type,
+        innhold = fødselsdato?.let { objectMapper.valueToTree(Person(fødselsdato = it)) }
+            ?: objectMapper.createObjectNode()
+    )
+
+    fun byggBostatusgrunnlag(data: Beregningskontekst): List<GrunnlagDto> {
+        fun nyttBostatusgrunnlag(referanse: String, bostatus: Bostatuskode, gjelderBarnReferanse: String?, gjelderReferanse: String? = null) =
+            GrunnlagDto(
+                referanse = referanse,
+                type = Grunnlagstype.BOSTATUS_PERIODE,
+                innhold = objectMapper.valueToTree(
+                    (gjelderReferanse ?: gjelderBarnReferanse)?.let {
+                        BostatusPeriode(
+                            bostatus = bostatus,
+                            periode = ÅrMånedsperiode(YearMonth.now(), null),
+                            relatertTilPart = it,
+                            manueltRegistrert = true
+                        )
+                    }
+                ),
+                gjelderBarnReferanse = gjelderBarnReferanse,
+                gjelderReferanse = gjelderReferanse
+            )
+
+        fun barnGrunnlag(antall: Int, prefix: String, kode: Bostatuskode) = List(antall) { index ->
+            nyttBostatusgrunnlag(
+                referanse = "Bostatus_${prefix}_$index",
+                bostatus = kode,
+                gjelderBarnReferanse = "Barn_${prefix}_$index",
+                gjelderReferanse = Referanser.BIDRAGSPLIKTIG
+            )
+        }
+
+        val boforhold = if (data.barn.bidragstype == BidragsType.PLIKTIG) data.dto.dittBoforhold else data.dto.medforelderBoforhold
```

Comment:
Ta inn bidragspliktig her også?
```suggestion
        val boforholdBidragspliktig = if (data.barn.bidragstype == BidragsType.PLIKTIG) data.dto.dittBoforhold else data.dto.medforelderBoforhold
```

Reply:
Her er det ikke alltid det er boforhold til bidragspliktig, se i if´en